src/routes/transactions/mappers/common/transaction.mapper.ts

```diff
@@ -0,0 +1,142 @@
+import { Injectable } from '@nestjs/common';
+import {
+  isEthereumTransaction,
+  isModuleTransaction,
+  isMultisigTransaction,
+  Transaction as TransactionDomain,
+} from '../../../../domain/safe/entities/transaction.entity';
+import { Safe } from '../../../../domain/safe/entities/safe.entity';
+import { ModuleTransactionMapper } from '../module-transactions/module-transaction.mapper';
+import { MultisigTransactionMapper } from '../multisig-transactions/multisig-transaction.mapper';
+import { MultisigTransaction } from '../../../../domain/safe/entities/multisig-transaction.entity';
+import { ModuleTransaction } from '../../../../domain/safe/entities/module-transaction.entity';
+import { EthereumTransaction } from '../../../../domain/safe/entities/ethereum-transaction.entity';
+import { groupBy } from 'lodash';
+import { IncomingTransferMapper } from '../transfers/transfer.mapper';
+import { TransactionItem } from '../../entities/transaction-item.entity';
+import { TransactionDomainGroup } from '../../entities/transaction-history-group.entity';
+import { DateLabel } from '../../entities/date-label.entity';
+
+@Injectable()
+export class TransactionMapper {
+  constructor(
+    private readonly multisigTransactionMapper: MultisigTransactionMapper,
+    private readonly moduleTransactionMapper: ModuleTransactionMapper,
+    private readonly incomingTransferMapper: IncomingTransferMapper,
+  ) {}
+  async mapTransaction(
+    chainId: string,
+    transactionsDomain: TransactionDomain[],
+    safe: Safe,
+    timezoneOffset?: string,
+    offset?: number,
+  ): Promise<[TransactionItem | DateLabel]> {
+    const transactionList: any[] = [];
+
+    let prev_page_timestamp = 0;
+    if (offset !== undefined && offset > 0) {
+      // Get previous page label
+      const timestamp =
+        transactionsDomain[0].executionDate?.getTime() ??
+        (
+          transactionsDomain[0] as MultisigTransaction
+        ).submissionDate?.getTime();
+      if (timestamp !== undefined) {
+        prev_page_timestamp = this.getDayInMillis(timestamp, timezoneOffset);
+      } else {
+        throw Error('ExecutionDate cannot be null');
+      }
+      transactionsDomain = transactionsDomain.slice(1);
+    }
+    const transactionsDomainGroups = this.groupByDay(
+      transactionsDomain,
+      timezoneOffset,
+    );
+    transactionsDomainGroups.forEach((transactionGroup) => {
```

Comment:
There's a lot of nesting in this section too which makes it more difficult to read and follow the code.

My suggestion is to restructure this part. It seems that we are tackling different types so that's a good starting point for splitting these branches.

Reply:
Done 83b9483823373e2c0541a968e0412c38f6d2c766